test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1304 @@
+		}
+		res = v1.ResourceRequirements{Limits: lim, Requests: req}
+	}
+	if tcInfo.Allocations != nil {
+		alloc = make(v1.ResourceList)
+		if tcInfo.Allocations.CPUAlloc != "" {
+			alloc[v1.ResourceCPU] = resource.MustParse(tcInfo.Allocations.CPUAlloc)
+		}
+		if tcInfo.Allocations.MemAlloc != "" {
+			alloc[v1.ResourceMemory] = resource.MustParse(tcInfo.Allocations.MemAlloc)
+		}
+		if tcInfo.Allocations.ephStorAlloc != "" {
+			alloc[v1.ResourceEphemeralStorage] = resource.MustParse(tcInfo.Allocations.ephStorAlloc)
+		}
+
+	}
+	if tcInfo.CPUPolicy != nil {
+		cpuPol := v1.ContainerResizePolicy{ResourceName: v1.ResourceCPU, RestartPolicy: *tcInfo.CPUPolicy}
+		resizePol = append(resizePol, cpuPol)
+	}
+	if tcInfo.MemPolicy != nil {
+		memPol := v1.ContainerResizePolicy{ResourceName: v1.ResourceMemory, RestartPolicy: *tcInfo.MemPolicy}
+		resizePol = append(resizePol, memPol)
+	}
+	return res, alloc, resizePol
+}
+
+func initDefaultResizePolicy(containers []TestContainerInfo) {
+	noRestart := v1.NotRequired
+	setDefaultPolicy := func(ci *TestContainerInfo) {
+		if ci.CPUPolicy == nil {
+			ci.CPUPolicy = &noRestart
+		}
+		if ci.MemPolicy == nil {
+			ci.MemPolicy = &noRestart
+		}
+	}
+	for i := range containers {
+		setDefaultPolicy(&containers[i])
+	}
+}
+
+func makeTestContainer(tcInfo TestContainerInfo) (v1.Container, v1.ContainerStatus) {
+	cmd := "grep Cpus_allowed_list /proc/self/status | cut -f2 && sleep 1d"
+	res, alloc, resizePol := getTestResourceInfo(tcInfo)
+
+	tc := v1.Container{
+		Name:         tcInfo.Name,
+		Image:        imageutils.GetE2EImage(imageutils.BusyBox),
+		Command:      []string{"/bin/sh"},
+		Args:         []string{"-c", cmd},
+		Resources:    res,
+		ResizePolicy: resizePol,
+	}
+
+	tcStatus := v1.ContainerStatus{
+		Name:               tcInfo.Name,
+		AllocatedResources: alloc,
+	}
+	return tc, tcStatus
+}
+
+func makeTestPod(ns, name, timeStamp string, tcInfo []TestContainerInfo) *v1.Pod {
+	var testContainers []v1.Container
+	var podOS *v1.PodOS
+
+	for _, ci := range tcInfo {
+		tc, _ := makeTestContainer(ci)
+		testContainers = append(testContainers, tc)
+	}
+
+	podOS = &v1.PodOS{Name: v1.Linux}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+			Labels: map[string]string{
+				"time": timeStamp,
+			},
+		},
+		Spec: v1.PodSpec{
+			OS:            podOS,
+			Containers:    testContainers,
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+	return pod
+}
+
+func verifyPodResizePolicy(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	cMap := make(map[string]*v1.Container)
+	for i, c := range pod.Spec.Containers {
+		cMap[c.Name] = &pod.Spec.Containers[i]
+	}
+	for _, ci := range tcInfo {
+		gomega.Expect(cMap).Should(gomega.HaveKey(ci.Name))
+		c := cMap[ci.Name]
+		tc, _ := makeTestContainer(ci)
+		gomega.Expect(tc.ResizePolicy).To(gomega.Equal(c.ResizePolicy))
+	}
+}
+
+func verifyPodResources(pod *v1.Pod, tcInfo []TestContainerInfo) {
```

Comment:
ditto

Reply:
Will try and comeback with a commit GinkgoHelper, thanks for sharing

Reply:
Done